### PR TITLE
Add Characteristic User Description GATT descriptor

### PR DESCRIPTION
**Pull request by beaufortfrancois**
This PR adds Characteristic User Description GATT descriptor, fixes an issue when reading CCC GATT descriptors.

R: @g-ortuno

### CODE REVIEW

File: app/src/main/java/io/github/webbluetoothcg/bletestperipheral/HealthThermometerServiceFragment.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
+/*
```
Comment (g-ortuno) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/HealthThermometerServiceFragment.java]: Can you do this in a separate patch?

Comment (g-ortuno) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/HealthThermometerServiceFragment.java]: Can you do this in a separate patch?

File: app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java [hunk 2]
```diff
@@ -346,8 +366,25 @@ public void run() {
   ////// Bluetooth //////
   ///////////////////////
   public static BluetoothGattDescriptor getClientCharacteristicConfigurationDescriptor() {
-    return new BluetoothGattDescriptor(CLIENT_CHARACTERISTIC_CONFIGURATION_UUID,
-            (BluetoothGattDescriptor.PERMISSION_READ | BluetoothGattDescriptor.PERMISSION_WRITE));
+    BluetoothGattDescriptor descriptor = new BluetoothGattDescriptor(
+        CLIENT_CHARACTERISTIC_CONFIGURATION_UUID,
+        (BluetoothGattDescriptor.PERMISSION_READ | BluetoothGattDescriptor.PERMISSION_WRITE));
+    try {
```
Comment (g-ortuno) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java]: Why do we need to set the value?

File: app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java [hunk 3]
```diff
@@ -346,8 +366,25 @@ public void run() {
...
+    } finally {
+      return descriptor;
+    }
+  }
+
+  public static BluetoothGattDescriptor getCharacteristicUserDescriptionDescriptor(String defaultValue) {
+    BluetoothGattDescriptor descriptor = new BluetoothGattDescriptor(
+        CHARACTERISTIC_USER_DESCRIPTION_UUID,
+        (BluetoothGattDescriptor.PERMISSION_READ | BluetoothGattDescriptor.PERMISSION_WRITE));
+    try {
```
Comment (g-ortuno) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java]: Why do we need the try?

**Comment by g-ortuno** [comment]
Sorry for taking so long! I just back to my office.

File: app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java (hunk 3 above)
Comment (beaufortfrancois) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java]: We don't ;(

File: app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java (hunk 2 above)
Comment (beaufortfrancois) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java]: You need it because I've added the ability to read Descriptors while it wasn't possible before.
See https://github.com/WebBluetoothCG/ble-test-peripheral-android/pull/76/files#diff-9dee403cd7eefd5dbe7d5d5fe63816e2R178

**Comment by beaufortfrancois** [comment]
@g-ortuno I've addressed your feedback and submitted a separate patch for the Health Thermometer Service.

File: app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java (hunk 2 above)
Comment (g-ortuno) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java]: Right but we don't do this for characteristics right? So why do we need to do it for descriptors.?

Comment (beaufortfrancois) [inline app/src/main/java/io/github/webbluetoothcg/bletestperipheral/Peripheral.java]: This `setValue` is called in `getClientCharacteristicConfigurationDescriptor`. This is the way we initialize the CCCD. 

For characteristics, we set the characteristic value for instance in `setBatteryLevel` called in `createView`.

**Comment by g-ortuno** [comment]
lgtm

**Comment by beaufortfrancois** [comment]
@g-ortuno I've merged this PR. I'll wait for https://github.com/WebBluetoothCG/ble-test-peripheral-android/pull/77 to be reviewed as well before incrementing version.
